syllabus/curricula/plr/task_sampler.py

```diff
@@ -285,16 +368,54 @@ def sample(self, strategy=None):
                 if np.random.rand() > self.nu or not proportion_seen < 1.0:
                     return self._sample_replay_level()
 
-            # Otherwise, sample a new level
-            return self._sample_unseen_level()
+            # Otherwise, evaluate a new level
+            return self._evaluate_unseen_level()
```

Comment:
A few changes need to be made here. 
1. This function needs to return a task sampled from the buffer
2. The returned data from self.evaluate_unseen_level needs to be used to call update_with_episode_data
So maybe something like this would work:
```suggestion
            self.update_with_episode_data(self._evaluate_unseen_level())
            return self.sample(strategy=strategy)
```

This will recursively call the function until it randomly decides to sample from the replay buffer, which shouldn't take too many attempts.